apollo-federation/src/query_graph/build_query_graph.rs

```diff
@@ -1888,22 +1888,10 @@ impl FederatedQueryGraphBuilder {
                             followup_edge_weight.transition,
                             QueryGraphEdgeTransition::KeyResolution
                         ) {
-                            let Some(conditions) = &edge_weight.conditions else {
```

Comment:
previously we were failing when there are no conditions on key resolution edges -> should we continue doing this?

Reply:
Based on the _must_ language here:
https://github.com/apollographql/router/blob/d2a4b1cbd2756a021ac6f8d94f6cb5ee960e820f/apollo-federation/src/query_graph/mod.rs#L195-L198
I think we should error if it's missing.